pkg/ir/ir.go

```diff
@@ -0,0 +1,66 @@
+package ir
+
+import (
+	route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+)
+
+// Xds holds the intermediate representation of a Gateway and is
+// used by the xDS Translator to convert it into xDS resources
+type Xds struct {
+	Name string
+	// One or more HTTP or HTTPS listeners exposed by the gateway
+	Http []HttpListener
+}
+
+type HttpListener struct {
+	// Port on which the service can be expected to be accessed by clients.
+	Port uint32
+
+	// Hostnames (Host/Authority header value) with which the service can be expected to be accessed by clients.
+	Hostnames []string
+
+	// TLS certificate info. If omitted, the gateway will expose a plain text HTTP server.
+	Tls ServerTLSSettings
+
+	// Routing rules associated with HTTP traffic to the service.
+	Routes []HttpRoute
+}
+
+type HttpRoute struct {
+	// Match condition.
+	Matchers     []route.HeaderMatcher
+	Destinations []RouteDestination
+}
+
+type RouteDestination struct {
+	// FQDN or IP address of the backend service.
+	Host string
+	// The port on the service to forward the request to.
+	Port uint32
+	// Weight associated with this destination.
+	Weight uint32
+	// Priority of the destination.
```

Comment:
How is Priority realized by EG when running in Kube?

Reply:
this ties to https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/priority in envoy allowing the user to create backup routes etc
can rm this for now since its not present in Gateway API, but would be a great addition to it 
cc @youngnick